libraries/AP_Scripting/generator/src/main.c

```diff
@@ -1956,6 +1956,14 @@ void emit_argcheck_helper(void) {
   fprintf(source, "}\n\n");
 }
 
+void emit_not_supported_helper(void) {
+  fprintf(source, "static int not_supported_error(lua_State *L, int expected_arg_count, const char* name) {\n");
+  fprintf(source, "    char string[50];\n");
+  fprintf(source, "    strcpy(string, name);\n");
```

Comment:
`strncpy`, and `strncat` and make sure you don't overflow the 50 byte space + null terminate it.

Alternatively and probably better would be to snprintf() the whole thing:

```
  fprintf(source, "    snprintf(string, sizeof(string), \"%s not supported on this firmware\", name);\n");
```

This will add the null byte at the end for you and do all the formatting. (Also maybe consider replacing string with `error_msg` or similar)

Reply:
Done, thanks, saves a extra 40 bytes too